test/routes/src/server/routes/t_launchpad.js

```diff
@@ -569,6 +570,272 @@ describe('The Launchpad API endpoint', () => {
       });
     });
   });
+
+
+  describe('get snap builds route', () => {
+    const lp_snap_user = 'test-snap-user';
+    const lp_snap_name = 'test-snap-name';
+
+    let lp_api_url;
+    let lp_snap_path;
+    let lp_builds_path;
+    let lp_snap_link;
+
+    before(() => {
+      lp_api_url = conf.get('LP_API_URL');
+      lp_snap_path = `/devel/~${lp_snap_user}/+snap/${lp_snap_name}`;
+      lp_builds_path = `${lp_snap_path}/builds`;
+
+      lp_snap_link = `${lp_api_url}${lp_snap_path}`;
+    });
+
+    context('when snap and builds are successfully fetched', () => {
+      let test_build;
+
+      beforeEach(() => {
+        test_build = {
+          'resource_type_link': `${lp_api_url}/devel/#snap_build`,
+          'self_link': `${lp_api_url}${lp_snap_path}/+build/12345`,
+        };
+
+        // when getting snap data from API (via self_link)
+        nock(lp_api_url)
+          .get(lp_snap_path)
+          .reply(200, {
+            name: lp_snap_name,
+            builds_collection_link: `${lp_api_url}${lp_builds_path}`
+          });
+
+        // when getting builds list (via builds_collection_link)
+        nock(lp_api_url)
+          .get(lp_builds_path)
+          .query({
+            'ws.start': 0,
+            'ws.size': 10
+          })
+          .reply(200, {
+            total_size: 1,
+            start: 0,
+            entries: [ test_build ]
+          });
+      });
+
+      afterEach(() => {
+        nock.cleanAll();
+      });
+
+      it('should return a 200 OK response', (done) => {
+        supertest(app)
+          .get('/launchpad/builds')
+          .query({ snap_link: lp_snap_link })
+          .expect(200, done);
+      });
+
+      it('should return a "success" status', (done) => {
+        supertest(app)
+          .get('/launchpad/builds')
+          .query({ snap_link: lp_snap_link })
+          .expect(hasStatus('success'))
+          .end(done);
+      });
+
+      it('should return body with "snap-builds-found" message', (done) => {
+        supertest(app)
+          .get('/launchpad/builds')
+          .query({ snap_link: lp_snap_link })
+          .expect(hasMessage('snap-builds-found'))
+          .end(done);
+      });
+
+      it('should return builds list in payload', (done) => {
+        supertest(app)
+          .get('/launchpad/builds')
+          .query({ snap_link: lp_snap_link })
+          .end((err, res) => {
+            const build = res.body.payload.builds[0];
+            // XXX bartaz
+            // Wanted to use `expect(build).toEqual(test_build)`
+            // but LP client adds more propeties to the object...
+            // also `expect(build).toContain(test_build)` doesnt work
+            // because `expect` fails to iterate over keys on the object
+            // for some reason (async maybe).
+            // To be investigated later...
+            expect(build.self_link).toEqual(test_build.self_link);
```

Comment:
@cjwatson @katiefenn One thing I noticed (may be worth remembering if you ever have similar issue) is that objects returned from LP client are not pure JSON objects, but have classes defined in `launchpad/resources.js`.

They have additional properties (not just what is returned by API) and for some reason `expect` has some trouble iterating over the keys in object when using `expect().toContain()`.

It's nothing really blocking ATM, but it may hit us again. It may be worth investigating when we have more time why it happens and if we can/need to do something about it.